tests/threads/testasynceventwiththreads.nim

```diff
@@ -0,0 +1,27 @@
+discard """
+output: '''
+runForever should throw ValueError, this is expected
+triggerCount: 1000
+'''
+"""
+
+import asyncDispatch, threadpool, os, random
+
+var triggerCount = 0
+var evs = newSeq[AsyncEvent]()
+
+proc threadTask(ev: AsyncEvent) =
+  sleep(rand(1000))
+  ev.trigger()
+
+for i in 0 ..< 1000:
+  var ev = newAsyncEvent()
+  evs.add ev
+  addEvent(ev, proc(fd: AsyncFD): bool {.gcsafe,closure.} = triggerCount += 1; true)
+  spawn(threadTask(ev))
+
+try:
+  runForever()
+except ValueError:
+  echo "runForever should throw ValueError, this is expected"
```

Comment:
Why is this expected? Add a comment in code to explain.

Reply:
Good question. I stole it from here (where it also doesn't have a comment to explain):
https://github.com/nim-lang/Nim/blob/devel/tests/async/t8982.nim

@narimiran is who you need to yell at for not commenting their code! /rant :-P

The answer is that `runOnce()` will throw an exception when it has no more pending operations:
https://github.com/nim-lang/Nim/blob/1e5a071c5af7a74c5d07faef8cf2ed3007e3bf9c/lib/pure/asyncdispatch.nim#L361

This is basically a hacky alternative to` drain()`
Remember I discussed this in this comment: https://github.com/nim-lang/Nim/pull/12371#discussion_r334256603

This test was the reason I started looking into `drain()` in the first place, and discovered #12371 

Once #12371 is merged I think I can refactor this test to use `drain()`